qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteringRule.java

```diff
@@ -602,27 +603,38 @@ public void stopBroker(final int nodeId) {
     }
   }
 
-  public void forceClusterToHaveNewLeader(final int expectedLeader) {
-    final var previousLeader = getCurrentLeaderForPartition(1);
-    if (previousLeader.getNodeId() == expectedLeader) {
+  public void forceClusterToHaveNewLeader(final int partitionId, final int expectedLeaderId) {
+    final var previousLeader = getCurrentLeaderForPartition(partitionId);
+    final var expectedLeader = brokers.get(expectedLeaderId);
+
+    if (previousLeader.getNodeId() == expectedLeaderId) {
       return;
     }
 
-    final var broker = brokers.get(expectedLeader);
-    final var atomix = broker.getBrokerContext().getClusterServices();
-    final MemberId nodeId = atomix.getMembershipService().getLocalMember().id();
-
-    final var raftPartition =
-        broker.getBrokerContext().getPartitionManager().getPartitionGroup().getPartitions().stream()
-            .filter(partition -> partition.members().contains(nodeId))
-            .filter(partition -> partition.id().id() == START_PARTITION_ID)
-            .map(RaftPartition.class::cast)
-            .findFirst()
-            .orElseThrow();
-
-    raftPartition.getServer().promote().join();
+    final var serverOfExpectedLeader =
+        ((RaftPartition)
+                expectedLeader
+                    .getBrokerContext()
+                    .getPartitionManager()
+                    .getPartitionGroup()
+                    .getPartition(partitionId))
+            .getServer();
+
+    Awaitility.await("Promote request is successful")
+        .pollInterval(Duration.ofMillis(500))
```

Comment:
:question: Any specifc reason for that poll  interval?

Reply:
Not at all, 500ms just felt like a reasonable interval before retrying :shrug:

Reply:
Hm ok I would just remove it, otherwise it looks like it is important :shrug: